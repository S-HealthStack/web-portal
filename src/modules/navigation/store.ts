import { createSelector } from '@reduxjs/toolkit';
import { matchPath } from 'react-router-dom';
import { createBrowserHistory, createMemoryHistory, Location } from 'history';

import { RootState } from 'src/modules/store/store';

export enum Path {
  Root = '/',

  SignIn = '/signin',
  AccountActivation = '/account/activation',
  AccountCreate = '/account/create',
  AccountConfirm = '/account/confirm',
  AccountVerification = '/email-verification',
  ForgotPassword = '/account/forgot',
  ForgotPasswordConfirm = '/account/forgot/confirm',
  ResetPassword = '/account/forgot/reset',
  ResetPasswordComplete = '/account/forgot/complete',

  CreateStudy = '/create-study',

  Overview = '/overview',
  OverviewSubject = '/overview/participants/:subjectId',
  TrialManagement = '/trial-management',
  TrialManagementSubject = '/trial-management/participants/:subjectId',
  TrialManagementEditSurvey = '/trial-management/surveys/:surveyId/edit',
  TrialManagementSurveyResults = '/trial-management/surveys/:surveyId/results',
  UserAnalytics = '/user-analytics',
  DataCollection = '/data-collection',
  DataCollectionSubject = '/data-collection/participants/:subjectId',
  StudySettings = '/study-settings',
}

export const makeHistory = () =>
  process.env.NODE_ENV !== 'test'
    ? createBrowserHistory({ basename: process.env.PUBLIC_PATH })
    : createMemoryHistory();

export const history = makeHistory();

const sectionPaths = [
  Path.Overview,
  Path.OverviewSubject,
  Path.TrialManagement,
  Path.UserAnalytics,
  Path.DataCollection,
  Path.StudySettings,
] as const;

export const sectionPathSelector = createSelector(
  (state: RootState) => state.router,
  (r) => sectionPaths.find((s) => matchPath((r.location as unknown as Location).pathname, s))
);
