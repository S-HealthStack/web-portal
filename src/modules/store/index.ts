export type { RootState, AppDispatch, AppThunk } from './store';
export { useAppDispatch, useAppSelector } from './hooks';
export * from './tools';
