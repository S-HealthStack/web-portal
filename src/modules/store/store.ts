/* eslint-disable import/no-import-module-exports */
import {
  Action,
  ThunkAction,
  configureStore,
  getDefaultMiddleware,
  ThunkDispatch,
} from '@reduxjs/toolkit';
import { routerMiddleware } from 'connected-react-router';

import { history } from 'src/modules/navigation/store';
import rootReducer from 'src/modules/store/reducers';

export { rootReducer };

export const store = configureStore({
  reducer: rootReducer,
  middleware: [...getDefaultMiddleware(), routerMiddleware(history)],
  devTools: process.env.NODE_ENV !== 'production',
});

if (module.hot) {
  // Enable Webpack hot module replacement for reducers
  module.hot.accept('./reducers', async () => {
    // eslint-disable-next-line import/no-useless-path-segments
    const nextRootReducer = (await import('../store/reducers')).default;
    store.replaceReducer(nextRootReducer);
  });
}

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = ThunkDispatch<RootState, unknown, Action>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
