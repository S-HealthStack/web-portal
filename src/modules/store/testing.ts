import { configureStore, ConfigureStoreOptions } from '@reduxjs/toolkit';
import { rootReducer, RootState } from './store';

export const createTestStore = (state: ConfigureStoreOptions<RootState>['preloadedState']) =>
  configureStore({
    reducer: rootReducer(),
    preloadedState: state,
  });
