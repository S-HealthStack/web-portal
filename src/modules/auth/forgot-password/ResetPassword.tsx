import React, { ChangeEvent, useCallback, useState } from 'react';
import styled from 'styled-components';
import useKey from 'react-use/lib/useKey';
import useSearchParam from 'react-use/lib/useSearchParam';

import { px } from 'src/styles';
import ScreenCenteredCard from 'src/modules/auth/common/ScreenCenteredCard';
import PasswordInputField from 'src/common/components/PasswordInputField';
import Button from 'src/common/components/Button';
import { SnackbarContainer } from 'src/modules/snackbar';

import {
  PasswordRequirements,
  RequirementItem,
  useCheckPassword,
} from '../common/PasswordRequirements';
import { useResetPassword } from './resetPassword.slice';
import ScreenHeader from '../common/ScreenHeader';
import ScreenWrapper from '../common/ScreenWrapper';
import ScreenContentWrapper from '../common/ScreenContentWrapper';

const Content = styled.div`
  width: ${px(448)};
  margin: auto;
  display: flex;
  flex-direction: column;
`;

const Header = styled(ScreenHeader)`
  margin: ${px(0)} auto ${px(36)};
`;

const InputsWrapper = styled.div`
  margin-top: ${px(3)};
  margin-bottom: ${px(10)};
  display: flex;
  align-items: stretch;
  flex-direction: column;
  row-gap: ${px(17)};
`;

const ResetPassword = () => {
  const email = useSearchParam('email') || '';
  const resetToken = useSearchParam('reset-token') || '';

  const [password, setPassword] = useState('');

  const { isLoading, resetPassword } = useResetPassword();

  // TODO: maybe later backend will provide user name
  const name = email.split('@')[0] || '';

  const [isPasswordPassed, requirements] = useCheckPassword({ password, name });

  const isDisabled = !email || !resetToken || !password || !isPasswordPassed;

  const handlePasswordChange = useCallback(
    (event: ChangeEvent<HTMLInputElement>) => setPassword(event.target.value),
    []
  );

  const handleClick = useCallback(() => {
    resetPassword({ email, password, resetToken });
  }, [resetPassword, email, password, resetToken]);

  useKey('Enter', handleClick, undefined, [handleClick]);

  return (
    <>
      <ScreenWrapper
        mediaMaxHeightToScrollY={704}
        mediaMaxWidthToScrollX={732}
        data-testid="reset-password"
      >
        <ScreenCenteredCard minWidth={800} width={55.556} ratio={557 / 800}>
          <ScreenContentWrapper>
            <Content>
              <Header>Reset Password</Header>
              <InputsWrapper>
                <PasswordInputField
                  name="password"
                  error={password && !isPasswordPassed}
                  withoutErrorText
                  label="New password"
                  value={password}
                  onChange={handlePasswordChange}
                  placeholder="Enter password"
                  data-testid="reset-rassword-screen-password-field"
                />
              </InputsWrapper>
              <PasswordRequirements>
                {requirements.map((requirement) => (
                  <RequirementItem
                    key={requirement.rule}
                    rule={requirement.rule}
                    passed={!password ? undefined : requirement.passed}
                    disabled={!password}
                  />
                ))}
              </PasswordRequirements>
              <Button
                disabled={isDisabled}
                onClick={handleClick}
                fill="solid"
                $loading={isLoading}
                data-testid="reset-password-button"
              >
                Reset password
              </Button>
            </Content>
          </ScreenContentWrapper>
        </ScreenCenteredCard>
      </ScreenWrapper>
      <SnackbarContainer useSimpleGrid />
    </>
  );
};

export default ResetPassword;
