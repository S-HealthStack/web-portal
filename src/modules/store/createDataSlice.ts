import { useCallback, useEffect } from 'react';
import _isEqual from 'lodash/isEqual';
import useMount from 'react-use/lib/useMount';
import {
  ActionCreatorWithoutPayload,
  ActionCreatorWithPayload,
  createSlice,
  PayloadAction,
} from '@reduxjs/toolkit';
import type { Reducer } from 'redux';

import applyDefaultApiErrorHandlers from 'src/modules/api/applyDefaultApiErrorHandlers';
import { SnackbarInstance } from 'src/modules/snackbar/snackbar.slice';
import useShowSnackbar from 'src/modules/snackbar/useShowSnackbar';
import { AppThunk, ErrorType, RootState, WithLoadableData } from 'src/modules/store';

import { useAppDispatch, useAppSelector } from './hooks';

type PublicDataSliceState<D> = WithLoadableData<D>;

type EmptyFetchArgs = null;

export type DataSliceState<D, A> = PublicDataSliceState<D> & {
  fetchArgs: A | EmptyFetchArgs;
  prevFetchArgs: A | EmptyFetchArgs;
  isLoadingSilently?: boolean;
};

type DataSliceOptions<D, N extends string, A> = {
  name: N;
  fetchData: (args: A, dispatch: unknown) => Promise<D>;
};

type FetchActionOptions = {
  force?: boolean;
  silent?: boolean;
  onError?: (e?: unknown) => void;
};

type HookParams<A, D> = {
  initialData?: D;
  fetchArgs: A | boolean;
  refetchSilentlyOnMount?: boolean;
};

export type ShowOnErrorSnackbarParams = Omit<SnackbarInstance, 'id' | 'hideTimeoutId'> & {
  onAction?: () => void;
};

type DataSliceReturnType<D, N, A> = {
  name: N;
  reducer: Reducer<DataSliceState<D, A>>;
  stateSelector: (s: RootState) => DataSliceState<D, A>;
  hook: (
    params?: HookParams<A, D>,
    snackBarParams?: ShowOnErrorSnackbarParams
  ) => DataSliceState<D, A> & {
    fetch: (args: A, opts?: FetchActionOptions) => Promise<void>;
    refetch: () => Promise<void>;
    reset: () => void;
  };
  actions: {
    reset: ActionCreatorWithoutPayload;
    setData: ActionCreatorWithPayload<{ data: D }>;
    loadingStarted: ActionCreatorWithPayload<{ fetchArgs: A }>;
    loadingSuccess: ActionCreatorWithPayload<{ data: D }>;
    loadingFailed: ActionCreatorWithPayload<ErrorType>;
    fetch: (args: A, opts?: FetchActionOptions) => AppThunk<Promise<void>>;
    refetch: () => AppThunk<Promise<void>>;
  };
};

const isFetchArgsEmpty = <D, A>(
  fetchArgs: DataSliceState<D, A>['fetchArgs']
): fetchArgs is EmptyFetchArgs => fetchArgs === null;

function createDataSlice<D extends object, N extends string, A = void>(
  options: DataSliceOptions<D, N, A>
): DataSliceReturnType<D, N, A> {
  const initialState: DataSliceState<D, A> = {
    fetchArgs: null,
    prevFetchArgs: null,
  };

  const slice = createSlice({
    name: options.name,
    initialState,
    reducers: {
      setData(state, action: PayloadAction<{ data: D }>) {
        return Object.assign(state, {
          data: action.payload.data,
        });
      },
      loadingStarted(state, action: PayloadAction<{ fetchArgs: A; silent?: boolean }>) {
        return Object.assign(state, {
          isLoading: !action.payload.silent,
          isLoadingSilently: !!action.payload.silent,
          fetchArgs: action.payload.fetchArgs,
          prevFetchArgs: { ...state.fetchArgs },
          error: undefined,
        });
      },
      loadingSuccess(state, action: PayloadAction<{ data: D }>) {
        return Object.assign(state, {
          isLoading: false,
          isLoadingSilently: false,
          error: undefined,
          data: action.payload.data,
          prevFetchArgs: { ...state.fetchArgs },
        });
      },
      loadingFailed(state, action: PayloadAction<ErrorType>) {
        return Object.assign(state, {
          isLoading: false,
          isLoadingSilently: false,
          error: String(action.payload),
          data: undefined,
          prevFetchArgs: { ...state.fetchArgs },
        });
      },
      reset(state) {
        state.isLoading = undefined;
        state.isLoadingSilently = undefined;
        state.data = undefined;
        state.error = undefined;

        return Object.assign(state, initialState);
      },
    },
  });

  const sliceStateSelector = (state: RootState) =>
    state[options.name as keyof RootState] as DataSliceState<D, A>;

  const fetchThunk =
    (args: A, opts?: FetchActionOptions): AppThunk<Promise<void>> =>
    async (dispatch, getState) => {
      if (!opts?.force) {
        const { fetchArgs: currentArgs } = sliceStateSelector(getState());

        // fetch only if this is a first fetch or args have been changed
        if (!isFetchArgsEmpty(currentArgs) && _isEqual(currentArgs, args)) {
          return;
        }
      }
      dispatch(slice.actions.loadingStarted({ fetchArgs: args, silent: opts?.silent }));
      try {
        const data = await options.fetchData(args, dispatch);

        const { fetchArgs: currentArgs } = sliceStateSelector(getState());
        // discard received data
        // if another fetch with different args performed while fetchData was in progress
        if (_isEqual(args, currentArgs)) {
          dispatch(slice.actions.loadingSuccess({ data }));
        }
      } catch (err) {
        const { fetchArgs: currentArgs } = sliceStateSelector(getState());
        if (_isEqual(args, currentArgs)) {
          console.error(err);
          dispatch(
            slice.actions.loadingFailed(
              (err instanceof Error ? err.message : undefined) ||
                `Failed to fetch data ${options.name}`
            )
          );

          if (!applyDefaultApiErrorHandlers(err, dispatch, false)) {
            opts?.onError?.(err);
          }
        }
      }
    };

  const refetchThunk = (): AppThunk<Promise<void>> => async (dispatch, getState) => {
    const { fetchArgs } = sliceStateSelector(getState());
    if (!isFetchArgsEmpty(fetchArgs)) {
      await dispatch(fetchThunk(fetchArgs, { force: true }));
    }
  };

  const refetchSilentlyIfRequiredThunk =
    (): AppThunk<Promise<void>> => async (dispatch, getState) => {
      const state = sliceStateSelector(getState());
      if (!state.isLoading && !state.isLoadingSilently && !isFetchArgsEmpty(state.fetchArgs)) {
        await dispatch(fetchThunk(state.fetchArgs, { force: true, silent: !state.error }));
      }
    };

  const useSliceHook = (params?: HookParams<A, D>, snackBarParams?: ShowOnErrorSnackbarParams) => {
    const dispatch = useAppDispatch();
    const sliceState = useAppSelector((state) => sliceStateSelector(state));
    const isStudyListLoading = useAppSelector((state) => state.studies.isLoading);
    const showSnackBar = useShowSnackbar();

    const onError = () => {
      if (snackBarParams) {
        showSnackBar(snackBarParams);
      }
    };

    const hookResult = {
      fetch: useCallback(
        (args: A, opts?: FetchActionOptions) => dispatch(fetchThunk(args, { ...opts, onError })),
        // eslint-disable-next-line react-hooks/exhaustive-deps
        [dispatch]
      ),
      refetch: useCallback(() => dispatch(refetchThunk()), [dispatch]),
      reset: useCallback(() => dispatch(slice.actions.reset()), [dispatch]),
      ...sliceState,
      isLoading: sliceState.isLoading || isStudyListLoading,
    };

    const hasParams = !!params;
    useEffect(() => {
      if (hasParams && typeof params.fetchArgs !== 'boolean') {
        dispatch(fetchThunk(params.fetchArgs, { onError }));
      }
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [dispatch, hasParams, params?.fetchArgs]);

    useEffect(() => {
      if (
        !params?.initialData ||
        (!isFetchArgsEmpty(sliceState.fetchArgs) &&
          _isEqual(sliceState.fetchArgs, params?.fetchArgs))
      ) {
        return;
      }

      dispatch(slice.actions.setData({ data: params.initialData }));
    }, [dispatch, params?.fetchArgs, params?.initialData, sliceState.data, sliceState.fetchArgs]);

    useMount(() => {
      if (params?.refetchSilentlyOnMount) {
        dispatch(refetchSilentlyIfRequiredThunk());
      }
    });

    return hookResult;
  };

  return {
    name: options.name,
    reducer: slice.reducer,
    stateSelector: sliceStateSelector,
    hook: useSliceHook,
    actions: {
      ...slice.actions,
      fetch: fetchThunk,
      refetch: refetchThunk,
    },
  };
}

export default createDataSlice;
