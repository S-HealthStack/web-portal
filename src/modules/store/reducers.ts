import { combineReducers } from '@reduxjs/toolkit';
import { connectRouter } from 'connected-react-router';
import type { Reducer } from 'redux';

import authReducer from 'src/modules/auth/auth.slice';
import studySettingsReducer from 'src/modules/study-settings/studySettings.slice';
import snackbarReducer from 'src/modules/snackbar/snackbar.slice';
import overviewReducers from 'src/modules/overview/overview.slice';
import studiesReducer from 'src/modules/studies/studies.slice';
import studyManagementReducer from 'src/modules/study-management/studyManagement.slice';
import dataCollectionReducer from 'src/modules/data-collection/dataCollection.slice';
import sidebarReducer from 'src/modules/main-layout/sidebar/sidebar.slice';
import fileUploadReducer from 'src/modules/file-upload/fileUpload.slice';
import { history } from 'src/modules/navigation/store';
import { DataSliceState } from 'src/modules/store/createDataSlice';
import forgotPasswordReducers from 'src/modules/auth/forgot-password/forgotPassword.slice';
import resetPasswordReducers from 'src/modules/auth/forgot-password/resetPassword.slice';
import objectStorageReducers from 'src/modules/object-storage/objectStorage.slice';

const createReducer = <T, V>(asyncReducers?: { [key: string]: Reducer<DataSliceState<T, V>> }) =>
  combineReducers({
    ...authReducer,
    router: connectRouter(history),
    snackbar: snackbarReducer,
    dataCollection: dataCollectionReducer,
    sidebar: sidebarReducer,
    ...studySettingsReducer,
    ...overviewReducers,
    ...studyManagementReducer,
    ...forgotPasswordReducers,
    ...resetPasswordReducers,
    ...fileUploadReducer,
    ...asyncReducers,
    ...objectStorageReducers,
    ...studiesReducer,
  });

export default createReducer;
