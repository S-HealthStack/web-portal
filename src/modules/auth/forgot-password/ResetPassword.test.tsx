import React from 'react';
import 'jest-styled-components';
import { act, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import theme from 'src/styles/theme';
import { makeHistory, Path } from 'src/modules/navigation/store';
import { ThemeProvider } from 'styled-components/';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { matchPath } from 'react-router-dom';

import ResetPassword from 'src/modules/auth/forgot-password/ResetPassword';
import { makeStore } from 'src/modules/store/store';

describe('ResetPassword', () => {
  let store: ReturnType<typeof makeStore>;
  let history: ReturnType<typeof makeHistory>;

  beforeEach(() => {
    history = makeHistory();
    store = makeStore(history);
  });

  let search = '';
  const mockSearch = (s: string) => {
    search = s;
    Object.defineProperty(window, 'location', {
      configurable: true,
      enumerable: true,
      get: () => new URL(`http://test.com/?${search}`),
    });
  };

  it('should reset password', async () => {
    const email = 'hello@samsung.com';
    mockSearch(`email=${email}&reset-token=123`);

    await act(() => {
      render(
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <ResetPassword />
            </ConnectedRouter>
          </Provider>
        </ThemeProvider>
      );
    });

    expect(await screen.findByTestId('reset-password')).toMatchSnapshot();

    const passwordField = await screen.findByTestId('reset-rassword-screen-password-field');
    const passwordValue = 'Password123456789!';
    await userEvent.type(passwordField, passwordValue);
    expect(passwordField).toHaveValue(passwordValue);

    const resetPasswordButton = await screen.findByTestId('reset-password-button');

    await userEvent.click(resetPasswordButton);

    expect(
      matchPath(history.location.pathname, {
        path: Path.ResetPasswordComplete,
        exact: true,
      })
    ).not.toBeNull();
  });

  it('[NEGATIVE] should prevent click if data is broken', async () => {
    mockSearch(`email=&reset-token=`);

    await act(() => {
      render(
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <ResetPassword />
            </ConnectedRouter>
          </Provider>
        </ThemeProvider>
      );
    });

    expect(await screen.findByTestId('reset-password')).toBeInTheDocument();

    const passwordField = await screen.findByTestId('reset-rassword-screen-password-field');
    const passwordValue = 'Password123456789!';
    await userEvent.type(passwordField, passwordValue);
    expect(passwordField).toHaveValue(passwordValue);

    const resetPasswordButton = await screen.findByTestId('reset-password-button');

    expect(resetPasswordButton).toBeDisabled();
  });
});
