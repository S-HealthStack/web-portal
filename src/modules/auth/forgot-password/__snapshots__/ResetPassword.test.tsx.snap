// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword should reset password 1`] = `
.c1 {
  color: #474747;
  min-width: 50rem;
  width: 55.556vw;
  min-height: 34.8125rem;
  height: 38.680865vw;
  max-height: 90vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: auto;
  background-color: #FFFFFF;
  box-shadow: 0 0 0.125rem 0rem rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c13 {
  background: none;
  border: none;
  padding: 0;
  height: 1.5rem;
  width: 1.5rem;
}

.c13 svg {
  fill: #4475E3;
}

.c13:active,
.c13:enabled:hover {
  cursor: pointer;
}

.c13:active svg,
.c13:enabled:hover svg {
  fill: #4475E3;
}

.c13:disabled svg {
  fill: #DBDBDB;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.c12 > svg {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 2rem);
}

.c11 {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  box-sizing: border-box;
  height: 3.5rem;
  width: 100%;
  margin: 0;
  padding: 1rem;
  background-color: #F7F8FA;
  border: 0.0625rem solid #F7F8FA;
  border-radius: 0.25rem;
  -webkit-transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  transition: border 300ms cubic-bezier(0.45,0.05,0,1);
  caret-color: #4475E3;
}

.c11:hover:enabled {
  border-color: #2F65E0;
}

.c11:active:enabled,
.c11:focus-visible {
  outline: none;
  border-color: #4475E3;
}

.c11:disabled {
  color: rgba(0,0,0,0.38);
  border-color: transparent;
  background-color: #f8f8f8;
}

.c11::-webkit-input-placeholder {
  color: #808080;
}

.c11::-moz-placeholder {
  color: #808080;
}

.c11:-ms-input-placeholder {
  color: #808080;
}

.c11::placeholder {
  color: #808080;
}

.c11:disabled::-webkit-input-placeholder {
  color: #DBDBDB;
}

.c11:disabled::-moz-placeholder {
  color: #DBDBDB;
}

.c11:disabled:-ms-input-placeholder {
  color: #DBDBDB;
}

.c11:disabled::placeholder {
  color: #DBDBDB;
}

.c11:-webkit-autofill,
.c11:-webkit-autofill:hover,
.c11:-webkit-autofill:focus {
  border: 0.0625rem solid #4475E3;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
  background-color: #F7F8FA !important;
  background-image: unset !important;
  -webkit-text-fill-color: #838383;
}

.c11:-webkit-autofill:focus {
  border-color: #4475E3;
}

.c9 {
  max-height: 3.5rem;
}

.c9:hover .c10:enabled {
  border-color: #2F65E0;
}

.c8 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  gap: 0.5rem;
  height: 1.125rem;
}

.c14 {
  position: relative;
  bottom: 2.6875rem;
  left: calc(100% - 2rem);
}

.c14:hover {
  cursor: pointer;
}

.c25 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  pointer-events: none;
}

.c23 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  overflow: hidden;
  -webkit-transition-duration: 400ms;
  transition-duration: 400ms;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 3rem;
  border: none;
  background-color: #4475E3;
}

.c23 > div {
  font-family: Inter,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
  font-style: normal;
}

.c23 > div svg {
  margin-right: 0.25rem;
}

.c23 > div {
  color: #FFFFFF;
}

.c23 > div svg {
  fill: #FFFFFF;
}

.c23:hover:enabled {
  background-color: #2F65E0;
}

.c23:disabled {
  background-color: #B3C6F1;
}

.c23:focus-visible:enabled {
  border: 0.125rem solid #FFFFFF;
  background-color: #4475E3;
  outline: 0.125rem solid #C7D6F7;
}

.c24 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c18 {
  display: inline-grid;
  gap: 0.25rem;
  grid-template-columns: 2.5rem 1fr;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  cursor: unset !important;
}

.c21 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #838383;
  -webkit-transition: color 300ms cubic-bezier(0.45,0.05,0,1);
  transition: color 300ms cubic-bezier(0.45,0.05,0,1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c20 {
  width: 2.5rem;
  height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c17 {
  display: inline-block;
}

.c17 input {
  display: none;
}

.c17 .c19 > svg > :nth-child(1) {
  -webkit-transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
  transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
  fill: #DBDBDB;
}

.c17 .c19 > svg > :nth-child(2) {
  -webkit-transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
  transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
}

.c15 {
  margin-top: 0.875rem;
  margin-bottom: 1.625rem;
}

.c16 {
  height: 2.5rem;
}

.c16 * {
  cursor: default;
}

.c22 {
  font-family: Inter,sans-serif;
  font-size: 0.75rem;
  line-height: 1.125rem;
  font-weight: 400;
  font-style: normal;
  color: #838383;
}

.c4 {
  font-family: Inter,sans-serif;
  font-size: 2rem;
  line-height: 2.625rem;
  font-weight: 600;
  font-style: normal;
  margin: 1rem auto 0.5rem;
  text-align: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  width: 28rem;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  margin: 0rem auto 2.25rem;
}

.c6 {
  margin-top: 0.1875rem;
  margin-bottom: 0.625rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: 1.0625rem;
}

@media (max-height:44rem) {
  .c0 {
    overflow-y: scroll;
  }
}

@media (max-width:45.75rem) {
  .c0 {
    overflow-x: scroll;
  }
}

<div
  class="c0"
  data-testid="reset-password"
>
  <div
    class="c1"
    width="55.556"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4 c5"
        >
          Reset Password
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
              data-testid="input-description"
            >
              New password
            </div>
            <div
              class="c9"
            >
              <input
                class="c10 c11"
                data-testid="reset-rassword-screen-password-field"
                name="password"
                placeholder="Enter password"
                type="password"
                value=""
              />
              <div
                class="c12"
              >
                <button
                  aria-label="Show Password"
                  class="c13 c14"
                  color="primary"
                >
                  <svg
                    data-testid="icon"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <div
              class="c17 "
              color="disabled"
              disabled=""
              kind="success"
              readonly=""
            >
              <label
                class="c18"
                for="1"
                kind="success"
              >
                <div
                  class="c19 c20"
                >
                  <svg />
                </div>
                <div
                  class="c21"
                  disabled=""
                >
                  <span
                    class="c22"
                  >
                    Be at least 12 characters in length
                  </span>
                </div>
              </label>
              <input
                checked=""
                data-testid="auth-signup-radio"
                disabled=""
                id="1"
                readonly=""
                type="checkbox"
              />
            </div>
          </div>
          <div
            class="c16"
          >
            <div
              class="c17 "
              color="disabled"
              disabled=""
              kind="success"
              readonly=""
            >
              <label
                class="c18"
                for="2"
                kind="success"
              >
                <div
                  class="c19 c20"
                >
                  <svg />
                </div>
                <div
                  class="c21"
                  disabled=""
                >
                  <span
                    class="c22"
                  >
                    Contains 1 upper case, 1 lower case, 1 number, and 1 special character
                  </span>
                </div>
              </label>
              <input
                checked=""
                data-testid="auth-signup-radio"
                disabled=""
                id="2"
                readonly=""
                type="checkbox"
              />
            </div>
          </div>
          <div
            class="c16"
          >
            <div
              class="c17 "
              color="disabled"
              disabled=""
              kind="success"
              readonly=""
            >
              <label
                class="c18"
                for="3"
                kind="success"
              >
                <div
                  class="c19 c20"
                >
                  <svg />
                </div>
                <div
                  class="c21"
                  disabled=""
                >
                  <span
                    class="c22"
                  >
                    Does not contain more than 3 identical characters in a row
                  </span>
                </div>
              </label>
              <input
                checked=""
                data-testid="auth-signup-radio"
                disabled=""
                id="3"
                readonly=""
                type="checkbox"
              />
            </div>
          </div>
          <div
            class="c16"
          >
            <div
              class="c17 "
              color="disabled"
              disabled=""
              kind="success"
              readonly=""
            >
              <label
                class="c18"
                for="4"
                kind="success"
              >
                <div
                  class="c19 c20"
                >
                  <svg />
                </div>
                <div
                  class="c21"
                  disabled=""
                >
                  <span
                    class="c22"
                  >
                    Does not contain username
                  </span>
                </div>
              </label>
              <input
                checked=""
                data-testid="auth-signup-radio"
                disabled=""
                id="4"
                readonly=""
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <button
          class="c23"
          data-testid="reset-password-button"
          disabled=""
          fill="solid"
          style="overflow: hidden; position: relative;"
        >
          <div
            class="c24"
            data-testid="content"
          >
            Reset password
          </div>
          <div
            class="c25"
            color="primaryPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
